orttraining/orttraining/python/training/onnxblock/loss/loss.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# loss.py
+
+import copy
+import onnx
+
+from ..graph import Graph
+from onnxruntime.training import onnxblock
+
+
+def _get_output_from_output_name(onnx_model, output_name):
+    """Returns the graph output given the output name"""
+
+    # Iterate over the graph outputs looking for output_name
+    for output in onnx_model.graph.output:
+        if output.name == output_name:
+            return output
+
+    raise LookupError("The provided output name {output_name} is not a graph output.")
+
+
+class MSELoss(Graph):
+    """MSELoss onnxblock for adding MSE loss to an onnx model."""
+
+    def __init__(self, reduction="mean"):
+        super(MSELoss, self).__init__()
+
+        self._reduction = reduction
+
+    def build(self, base_model, output_name, target_name="target"):
+        """Adds an MSELoss subgraph on top of the base_model."""
+
+        # deepcopy the base model so we don't inadvertently change the original
```

Comment:
I wanted to ensure that any changes made to the model were made locally and not to the original model which the user might want to use for other purposes.

The idea of the API was to use it programmatically as input/outputs as follows:

```py
inference_model = get_exported_model()
model_with_fc = linear(inference_model)
model_with_dropout = dropout(model_with_fc)
model_with_loss = mse_loss(model_with_dropout)
...
```

If we use a single model and manipulate it, we cannot offer guarantee that the original model can be used to compose other bigger models (since it has been manipulated).

Reply:
I made change so that we no longer need to maintain multiple copies of the model. All manipulations will happen on the same model. Please let me know how you feel about it.